### PR TITLE
feat: add build for platform=redhat with ubi based image

### PR DESCRIPTION
**Pull request by Revolyssup**
Fixes https://github.com/apache/apisix/issues/9454

Note: The path to rpm package needs to be clarified to make sure that the rpm package based on ubi is published on 
https://repos.apiseven.com/packages/redhat/apache-apisix-repo-1.0-1.noarch.rpm

### CODE REVIEW

File: redhat/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM registry.access.redhat.com/ubi8/ubi:8.6
+
+ARG APISIX_VERSION=3.3.0
+LABEL apisix_version="${APISIX_VERSION}"
+
+RUN yum install -y https://repos.apiseven.com/packages/redhat/apache-apisix-repo-1.0-1.noarch.rpm \
```
Comment (tao12345666333) [inline redhat/Dockerfile:23]: If you don't have this file, you can directly add the configuration file in the /etc/yum.repos.d directory.

File: Makefile [hunk 2]
```diff
@@ -69,6 +69,12 @@ build-on-centos:
 	$(ENV_DOCKER) build -t $(ENV_APISIX_IMAGE_TAG_NAME)-centos -f ./centos/Dockerfile centos
 	@$(call func_echo_success_status, "$@ -> [ Done ]")
 
+### build-on-redhat : Build apache/apisix:xx-redhat image
+.PHONY: build-on-centos
```
Comment (tao12345666333) [inline Makefile:73]: typo?

**Review by tao12345666333 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (Revolyssup) [reply Makefile:73]: yes. fixing it

**Comment by Revolyssup** [comment]
Overriden by https://github.com/apache/apisix-docker/pull/463

**Review by tao12345666333 — APPROVED** [review]

File: redhat/yum.repos.d/apache-apisix.repo [hunk 3]
```diff
@@ -0,0 +1,9 @@
+[release]
+name=Apache APISIX Repository for Redhat
+baseurl=https://repos.apiseven.com/packages/redhat/$releasever/$basearch
+skip_if_unavailable=False
+gpgcheck=1
+repo_gpgcheck=1
+gpgkey=https://repos.apiseven.com/KEYS
+enabled=1
+enabled_metadata=1
```
Comment (Sn0rt) [inline redhat/yum.repos.d/apache-apisix.repo:9]: new line EOF

File: redhat/yum.repos.d/openresty.repo [hunk 4]
```diff
@@ -0,0 +1,9 @@
+[openresty]
+name=Official OpenResty Open Source Repository for CentOS
+baseurl=https://openresty.org/package/centos/$releasever/$basearch
+skip_if_unavailable=False
+gpgcheck=1
+repo_gpgcheck=0
+gpgkey=https://openresty.org/package/pubkey.gpg
+enabled=1
+enabled_metadata=1
```
Comment (Sn0rt) [inline redhat/yum.repos.d/openresty.repo:9]: new line EOF

File: redhat/yum.repos.d/apache-apisix.repo (hunk 3 above)
Comment (Revolyssup) [reply redhat/yum.repos.d/apache-apisix.repo:9]: @Sn0rt You want me to add a new line EOF?

**Comment by tao12345666333** [comment]
https://github.com/apache/apisix-docker/blob/7dfe5840e7705ccee135f0220b12d369d52d0ca8/.github/workflows/apisix-docker-example-test.yaml you should change this file to run the test.

**Comment by Revolyssup** [comment]
> https://github.com/apache/apisix-docker/blob/7dfe5840e7705ccee135f0220b12d369d52d0ca8/.github/workflows/apisix-docker-example-test.yaml you should change this file to run the test.

@tao12345666333 done

**Review by tao12345666333 — APPROVED** [review]

**Review by Sn0rt — APPROVED** [review]

File: redhat/yum.repos.d/apache-apisix.repo (hunk 3 above)
Comment (leslie-tsang) [reply redhat/yum.repos.d/apache-apisix.repo:9]: Yes, Plz.

File: redhat/hooks/build [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+# hooks/build
```
Comment (leslie-tsang) [inline redhat/hooks/build:2]: Why we add a build hook here, it seems do the same thing as Makefile do.

Comment (Revolyssup) [reply redhat/hooks/build:2]: I was also confused about it but since it was present in centos and debian, I added it here. Should I remove?

Comment (leslie-tsang) [reply redhat/hooks/build:2]: Please check if it was necessary. IMO, it's not.

Comment (Revolyssup) [reply redhat/hooks/build]: @leslie-tsang  You're right. I couldn't find any usage of it in this repo so removed it.

**Review by leslie-tsang — APPROVED** [review]

File: redhat/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM registry.access.redhat.com/ubi8/ubi:8.6
+
+ARG APISIX_VERSION=3.3.0
+LABEL apisix_version="${APISIX_VERSION}"
+COPY ./yum.repos.d/apache-apisix.repo /etc/yum.repos.d/apache-apisix.repo
+COPY ./yum.repos.d/openresty.repo /etc/yum.repos.d/openresty.repo
```
Comment (soulbird) [inline redhat/Dockerfile:23]: We can download this file directly online: https://openresty.org/package/centos/openresty.repo

File: redhat/yum.repos.d/apache-apisix.repo [hunk 7]
```diff
@@ -0,0 +1,9 @@
+[release]
```
Comment (soulbird) [inline redhat/yum.repos.d/apache-apisix.repo:1]: Would it be better if I also upload this file to https://repos.apiseven.com/packages/redhat/ ？

Comment (Sn0rt) [reply redhat/yum.repos.d/apache-apisix.repo:1]: the CI loss. we this is another task.

File: redhat/Dockerfile (hunk 6 above)
Comment (Sn0rt) [reply redhat/Dockerfile:23]: https://github.com/api7/apisix-build-tools/issues/304

we will create a new ci to build this rpm package. so I think this modification postpone is acceptable

**Review by Sn0rt — APPROVED** [review]

**Pull request merged**